app/models/ingredient_name.rb

```diff
@@ -0,0 +1,5 @@
+class IngredientName < ActiveRecord::Base
+  belongs_to :ingredients
+
+  validates_uniqueness_of :recipe_ingredient_sub_name, scope: :ingredient_id
```

Comment:
I guess `:recipe_ingredient_sub_name` might not be always unique in general, but it should also be unique to a specific ingredient